### PR TITLE
Bump register interface to v0.3.1

### PR DESCRIPTION
**Pull request by andreaskuster**
This PR upgrades the `register_interface` to the lastest stable version (tested with Questasim)

### CODE REVIEW

**Comment by andreaskuster** [comment]
`On hold` until https://github.com/pulp-platform/rv_plic/pull/5 has been merged.

**Comment by andreaskuster** [comment]
The rv_plic PR has been merged and the tests pass. It is ready for review.

File: Makefile
Comment (niwis) [inline Makefile:104]: Indentation (use spaces instead of tabs)

**Review by niwis — COMMENTED** [review]
Thank you! I just spotted two whitespace errors. Could you also add the required changes to `ariane_peripherals_xilinx.sv`?

**Comment by andreaskuster** [comment]
Thanks for the input. I fixed the indentation and applied the changes to the `ariane_peripherals_xilinx.sv`

**Comment by niwis** [comment]
Great, thank you a lot :-) one last thing from my side: could you compile the FPGA image and boot Linux? Both the FPGA flow and PLIC are not thoroughly covered by the CI, so this would be an important feedback

**Comment by zarubaf** [comment]
Change LGTM, fine after the Linux boot shows now regression.

**Review by zarubaf — APPROVED** [review]

**Comment by andreaskuster** [comment]
I can confirm, the xilinx toolflow works (tested with `vivado 2020.2`) & boots Linux on the `Genesis 2` board.

**Pull request merged**
